packages/kbn-expandable-flyout/src/components/context.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import { noop } from 'lodash/fp';
+import type { FlyoutLayout } from '../models/layout';
+import type { FlyoutPanel } from '../models/panel';
+import { FlyoutState, initialFlyoutState } from '../models/state';
+import {
+  closeFlyout,
+  closeFlyoutLeftPanel,
+  closeFlyoutPreviewPanel,
+  closeFlyoutRightPanel,
+  openFlyout,
+  openFlyoutLeftPanel,
+  openFlyoutPreviewPanel,
+  openFlyoutRightPanel,
+  previousFlyoutPreviewPanel,
+  selectFlyoutLayout,
+} from '../utils/helpers';
+
+export interface ExpandableFlyoutContext {
+  /**
+   * Right, left and preview panels
+   */
+  panels: FlyoutLayout;
+  /**
+   * Open the flyout with left, right and/or preview panels
+   */
+  openFlyout: (panels: { left?: FlyoutPanel; right?: FlyoutPanel; preview?: FlyoutPanel }) => void;
+  /**
+   * Replaces the current right panel with a new one
+   */
+  openRightPanel: (panel: FlyoutPanel) => void;
+  /**
+   * Replaces the current left panel with a new one
+   */
+  openLeftPanel: (panel: FlyoutPanel) => void;
+  /**
+   * Add a new preview panel to the list of current preview panels
+   */
+  openPreviewPanel: (panel: FlyoutPanel) => void;
+  /**
+   * Closes right panel
+   */
+  closeRightPanel: () => void;
+  /**
+   * Closes left panel
+   */
+  closeLeftPanel: () => void;
+  /**
+   * Closes all preview panels
+   */
+  closePreviewPanel: () => void;
+  /**
+   * Go back to previous preview panel
+   */
+  previousPreviewPanel: () => void;
+  /**
+   * Close all panels and closes flyout
+   */
+  closeFlyout: () => void;
+}
+
+export const ExpandableFlyoutContext = createContext<ExpandableFlyoutContext>({
+  panels: {
+    left: {},
+    right: {},
+    preview: [],
+  },
+  openFlyout: noop,
+  openRightPanel: noop,
+  openLeftPanel: noop,
+  openPreviewPanel: noop,
+  closeRightPanel: noop,
+  closeLeftPanel: noop,
+  closePreviewPanel: noop,
+  closeFlyout: noop,
+  previousPreviewPanel: noop,
```

Comment:
is there a reason for having this like that instead of just using undefined & throwing when the context is not configured correctly?

Reply:
yes nice, done!